rcl/src/rcl/service.c

```diff
@@ -247,6 +247,8 @@ rcl_take_request(
 {
   RCUTILS_LOG_DEBUG_NAMED(ROS_PACKAGE_NAME, "Service server taking service request")
   const rcl_service_options_t * options = rcl_service_get_options(service);
+  RCL_CHECK_FOR_NULL_WITH_MSG(
+    options, "Failed to get service option", return RCL_RET_ERROR, rcl_get_default_allocator());
```

Comment:
Should the error message be "Failed to get service options" (note the trailing s)?

Same below.